File: database/notes/MultiThreading.java
// javac MultiThreading.java
// java MultiThreading

public class MultiThreading {
    
    static Integer count = 0;

    public static void increase() {
        for (int i = 0; i < 10000; i++) {
            count++;
            System.out.println(Thread.currentThread().getName() +  ": " + count);
        }
    }
    public static void main(String[] args) throws InterruptedException {
        Thread thread1 = new Thread(MultiThreading::increase);
        Thread thread2 = new Thread(MultiThreading::increase);

        thread1.start();
        thread2.start();

        thread1.join();
        thread2.join();
        System.out.println("Final value: " + count);

    }

}

File: database/notes/multi-threading.py
import threading
from time import sleep

class Number:
    def __init__(self, value: int) -> None:
        self.value = value
    def increase(self):
        self.value += 1
    def getValue(self):
        return self.value

count = Number(0)

def increase():
    for _ in range(10000):
        count.increase()
        print(str(threading.get_ident()) + ": " + str(count.getValue()))


t1 = threading.Thread(target=increase)
t2 = threading.Thread(target=increase)

t1.start()
t2.start()

t1.join()
t2.join()

print("Final value: " + str(count.getValue()))